jinahub/indexers/merger/MatchMerger/match_merger.py

```diff
@@ -0,0 +1,35 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import List, Tuple
+
+from jina import Executor, requests, DocumentArray
+
+
+class MatchMerger(Executor):
+    """
+    The MatchMerger merges the results of shards by appending all matches..
+
+    :param default_traversal_paths: traverse path on docs, e.g. ['r'], ['c']
+    :param args: additional arguments
+    :param kwargs: additional key value arguments
+    """
+
+    def __init__(self, default_traversal_paths: Tuple[str, ...] = ('r',), **kwargs):
+        super().__init__(**kwargs)
+        self.default_traversal_paths = default_traversal_paths
+
+    @requests
+    def merge(self, docs_matrix: List[DocumentArray], parameters: dict, **kwargs):
+        traversal_paths = parameters.get('traversal_paths', self.default_traversal_paths)
+        results = {}
+        for docs in docs_matrix:
+            self._merge_shard(results, docs, traversal_paths)
```

Comment:
What about some sorting of matches by score here?

Reply:
If we add sorting remember to pass the top_k parameter to sort only up to that level. And pass the name of the score to be based on and wether is distance or similarity/relevance

Reply:
Nice idea.
I would like to not support top-k for now and also let the ranker sort the results.
Just to keep this PR lean.